web/src/containers/ApdStateProfileMedicaidOffice.test.js

```diff
@@ -1,83 +1,148 @@
-import { shallow } from 'enzyme';
+import { mount } from 'enzyme';
 import React from 'react';
-import sinon from 'sinon';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
 
-import {
-  plain as StateProfile,
-  mapStateToProps,
-  mapDispatchToProps
-} from './ApdStateProfileMedicaidOffice';
-import { updateApd } from '../actions/apd';
+import { UPDATE_APD } from '../actions/apd';
+import { EDIT_APD } from '../actions/editApd';
+
+import StateProfile from './ApdStateProfileMedicaidOffice';
+
+const mockStore = configureStore([thunk]);
 
 describe('apd state profile, Medicaid office component', () => {
-  const props = {
-    stateProfile: {
-      medicaidDirector: {
-        name: 'name',
-        email: 'email',
-        phone: 'phone'
-      },
-      medicaidOffice: {
-        address1: 'address 1',
-        address2: 'address 2',
-        city: 'city',
-        state: 'state',
-        zip: 'zip'
+  const store = mockStore({
+    apd: {
+      data: {
+        stateProfile: {
+          medicaidDirector: {
+            name: 'name',
+            email: 'email',
+            phone: 'phone'
+          },
+          medicaidOffice: {
+            address1: 'address 1',
+            address2: 'address 2',
+            city: 'city',
+            state: 'state',
+            zip: 'zip'
+          }
+        }
       }
     },
-    updateApd: sinon.spy()
-  };
+    user: {
+      data: {
+        state: { id: 'mn' }
+      }
+    }
+  });
 
   beforeEach(() => {
-    props.updateApd.resetHistory();
+    store.clearActions();
   });
 
   test('renders correctly', () => {
-    const localProps = JSON.parse(JSON.stringify(props));
-
-    // all good
     expect(
-      shallow(<StateProfile {...localProps} updateApd={props.updateApd} />)
-    ).toMatchSnapshot();
-
-    // name length validation
-    localProps.stateProfile.medicaidDirector.name = 'a';
-    expect(
-      shallow(<StateProfile {...localProps} updateApd={props.updateApd} />)
+      mount(
+        <Provider store={store}>
```

Comment:
We also have to wrap the component in a `<Provider>` in order to give it the store that it needs access to. This thing is the reason we need to do a full mount - a shallow render would give us a `<Connected>` thing and inside it, which is what we really care about. A full mount gets us that whole thing.